### PR TITLE
Feat/main page video block

### PR DESCRIPTION
**Pull request by AziAlex**
Screenshot_31
Screenshot_30

### CODE REVIEW

**Review by AziAlex — COMMENTED** [review]
as

File: src/app/style/global.scss [hunk 1]
```diff
@@ -13,6 +13,13 @@ a {
   cursor: pointer;
 }
 
+p {
+  font-size: 16px;
+  font-weight: 400;
+  line-height: 150%;
+  color: #787A80;
```
Comment (theCapnCoder) [inline src/app/style/global.scss:20]: better to use global colors variables

File: src/feature/main-slider/style.scss [hunk 2]
```diff
@@ -1,22 +1,33 @@
...
-    width: 600px;
+    width: 100%;
+    max-width: 600px;
     position: absolute;
     z-index: 10;
+    left: calc(50% - 620px);
     bottom: 12%;
-    left: 18%;
-
     display: grid !important;
```
Comment (theCapnCoder) [inline src/feature/main-slider/style.scss:16]: It is good practice to never use !important

Can we rewrite this?

File: src/page/main-page/ui/video-block/index.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import Wrap from '@/shared/ui/wrap'
+import Image from 'next/image'
+import img from '/public/main-page/video-block.jpg'
+import Icon from '@/shared/svg'
+
+import cl from './style.module.scss'
+import UseResizeVideo from '@/page/main-page/ui/video-block/lib/hooks'
+
+const VideoBlock = () => {
+  const [videoWatch, setVideoWatch] = useState<boolean>(false)
```
Comment (theCapnCoder) [inline src/page/main-page/ui/video-block/index.tsx:15]: we can't use static typing, TS determines the type itself with dynamic typing

File: src/page/main-page/ui/video-block/index.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  const [videoConfig, setVideoConfig] = useState({
```
Comment (theCapnCoder) [inline src/page/main-page/ui/video-block/index.tsx:16]: Good example for dynamic typing

File: src/page/main-page/ui/video-block/index.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    height: '600',
+    width: '1065',
+    playerVars: {
+      autoplay: 1,
+    },
+  })
+
+  useEffect(() => {
+    if (videoWatch) {
+      document.body.style.overflow = 'hidden'
```
Comment (theCapnCoder) [inline src/page/main-page/ui/video-block/index.tsx:26]: We shouldn't refer to the document in react but I don't know if we can fix that in this case

**Review by AziAlex — COMMENTED** [review]
ы

File: src/page/main-page/ui/video-block/index.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            <div className={cl.play} onClick={() => setVideoWatch(true)}>
+              <Icon name='play' className={cl.play} />
+            </div>
+          </div>
+        </div>
+      </Wrap>
+      {videoWatch && (
+        <div
+          className={cl.videoWrap}
+          onClick={() => setVideoWatch(false)}
```
Comment (theCapnCoder) [inline src/page/main-page/ui/video-block/index.tsx:49]: if we are making a change to the state based on previous values it is better to use a callback

File: src/page/main-page/ui/video-block/lib/hooks.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
+import { useEffect, FC, useState } from 'react'
+
+interface IProps {
+  setVideoConfig: (config: any) => void
```
Comment (theCapnCoder) [inline src/page/main-page/ui/video-block/lib/hooks.tsx:4]: try not to use type "any"

File: src/page/main-page/ui/video-block/lib/hooks.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
+import { useEffect, FC, useState } from 'react'
+
+interface IProps {
```
Comment (theCapnCoder) [inline src/page/main-page/ui/video-block/lib/hooks.tsx:3]: we need to discuss what interface naming style we will use

File: src/page/main-page/ui/video-block/lib/hooks.tsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+    handleResize()
+    window.addEventListener('resize', handleResize)
+
+    return () => {
+      window.removeEventListener('resize', handleResize)
+    }
+  }, [setVideoConfig, width])
+
+  return <></>
```
Comment (theCapnCoder) [inline src/page/main-page/ui/video-block/lib/hooks.tsx:47]: Better return "null"

File: src/shared/svg/index.tsx [hunk 10]
```diff
@@ -15,7 +15,7 @@ import {
   IconCross,
   IconLike,
   IconSlippers,
-  IconHand,
+  IconHand, IconPlay,
```
Comment (theCapnCoder) [inline src/shared/svg/index.tsx:18]: Apply autoformat

File: src/shared/svg/ui/additional/IconPlay.tsx [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import React, { FC } from 'react'
+
+export const IconPlay: FC<{ className: string }> = ({ className }) => (
+  <svg className={className} viewBox='0 0 80 80' fill='none'>
```
Comment (theCapnCoder) [inline src/shared/svg/ui/additional/IconPlay.tsx:4]: We need to check where disappear xmlns="http://www.w3.org/2000/svg"

File: src/shared/svg/ui/additional/IconPlay.tsx [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+    <circle cx='40' cy='40' r='40' fill='#FF5A30' />
+    <path
+      d='M51.5 39.134C52.1667 39.5189 52.1667 40.4811 51.5 40.866L35 50.3923C34.3333 50.7772 33.5 50.2961 33.5 49.5263L33.5 30.4737C33.5 29.7039 34.3333 29.2228 35 29.6077L51.5 39.134Z'
+      fill='white' />
```
Comment (theCapnCoder) [inline src/shared/svg/ui/additional/IconPlay.tsx:8]: Maybe we need to delete this prop?

**Review by AziAlex — COMMENTED** [review]
completed fix

**Pull request merged**